### PR TITLE
Refactor for Database settings and added Integration Tests

### PR DESCRIPTION
**Pull request by nickryand**
I have added the ability to update the database settings written about in the Nexpose Admin guide. During that addition, I took the time to refactor pieces in an effort to simplify the cookbook and attributes. 

I also added Inspec integration tests and additional ChefSpec tests. 

There are a few things to note:
* I have not tested the changes on Windows
* The Linux recipe requires a start -> stop -> start sequence when running in Console mode in order to properly handle the database updates.
* The Inspec busser is still changing so there is a good chance the tests I have will break and need to be maintained.

### CODE REVIEW

**Comment by rhass-r7** [comment]
Thanks for submitting this PR!

I am working on updating the PR builder as well as testing the cookbook against our internal automation to make sure it doesn't break anything for us before we can land this.

**Comment by nickryand** [comment]
Awesome. Let me know if anything needs to be changed.

**Comment by rhass-r7** [comment]
I did a bunch of work to get the travis PR builder in a useful state. It seems to be working pretty well, and as a test I merged your code into a branch to see what would happen. As result, the travis build found a valid (but tiny) issue.   Specifically the inspec gem is missing from the Gemfile. 

https://travis-ci.org/rapid7-cookbooks/nexpose/builds/102280374

If you merge master, or the `PR#24_Merge-Test` branch into your `test-fixes` branch, the merge and any changes you push onto your branch should be properly tested in travis/ec2 for Windows, CentOS 7, and Ubuntu 12.04, and 14.04. This should allow us to let the machines do the work for us.

**Comment by nickryand** [comment]
rebased against master.

**Comment by nickryand** [comment]
Looks like there is something up with the ssh key decryption.

**Comment by rhass-r7** [comment]
Damn... seems I was a bit presumptuous about the PR builds.  

"Secure environment variables are enabled for pull requests as long as the origin of the pull request and the target repository are the same." https://blog.travis-ci.com/2013-06-10-secure-env-in-pull-requests/

"... we do not provide these values to untrusted builds, triggered by pull requests from another repository."
https://docs.travis-ci.com/user/environment-variables/#Defining-Variables-in-Repository-Settings


So basically, they only support it for PRs which are off a branch. It makes sense, as the attack vector is someone can submit a PR which cats the encrypted file(s) or prints the decrypted environment variables.  No matter, I will merge the changes into the `PR#24_Merge-Test` branch so we can get some results. Sorry about that.

**Comment by nickryand** [comment]
@rhass-r7 how did the tests look?

**Comment by rhass-r7** [comment]
So right now we have two issues, one being I had to push out a fix to the init scripts, which looks like it created a bunch of merge conflicts. The other, is the component_type changes. While they make a lot of sense, we have a lot of stuff internally which we need to refactor to accept that part.

If you can separate the changes around component type, we can at least land the bulk of the work for the postgresql tuning as long ast it remains and optional recipe which people can choose to include in their runlists.

**Comment by nickryand** [comment]
I can rebase and make the changes. 

For component_type, the end result is you need it in the config file or you need an attribute on the node object saying what time of component it is?
